uaclient/apt_news.py

```diff
@@ -88,6 +100,49 @@ def do_selectors_apply(
         if selectors.pro != _is_attached(cfg).is_attached:
             return False
 
+    if selectors.architectures is not None:
+        if system.get_dpkg_arch() not in selectors.architectures:
+            return False
+
+    if selectors.packages is not None:
+        installed_packages = get_installed_packages()
+        installed_packages_names = [
+            package.name for package in installed_packages
+        ]
+        package_matched = False
+
+        for package in selectors.packages:
+            package_name, version_operator, package_version = package
+            if package_name in installed_packages_names:
+                installed_package = installed_packages[
+                    installed_packages_names.index(package_name)
+                ]
+                version_comparison = version_compare(
+                    installed_package.version, package_version
+                )
+                if (
+                    (version_operator == "==" and version_comparison == 0)
+                    or (
+                        version_operator == "<="
+                        and (
+                            version_comparison == 0 or version_comparison == -1
+                        )
+                    )
+                    or (
+                        version_operator == ">="
+                        and (
+                            version_comparison == 0 or version_comparison == 1
+                        )
+                    )
+                    or (version_operator == "<" and version_comparison == -1)
+                    or (version_operator == ">" and version_comparison == 1)
+                ):
```

Comment:
Style suggestion
```suggestion
                if any(
                    [
                        (version_comparison == 0 and version_operator in ["==", "<=", ">="]),
                        (version_comparison < 0 and version_operator in ["<", "<="]),
                        (version_comparison > 0 and version_operator in [">", ">="]),
                    ]
                ):
```

Reply:
Looks much better than the original change haha, thanks @orndorffgrant